content/en/platform/corda/5.1/deploying-operating/vnodes/upgrade-cpi.md

```diff
@@ -13,18 +13,18 @@ section_menu: corda51
 
 # Upgrading a CPI
 
-To upgrade a virtual node's {{< tooltip >}}CPI{{< /tooltip >}}, use the PUT method of the <a href ="../../reference/rest-api/openapi.html#tag/Virtual-Node-API/operation/put_virtualnode__virtualnodeshortid__cpi__targetcpifilechecksum_">`api/v1/virtualnode/<virtualnodeshortid>/cpi/<targetcpifilechecksum>` endpoint </a>:
+You can upgrade a virtual node's {{< tooltip >}}CPI{{< /tooltip >}}, using the PUT method of the <a href ="../../reference/rest-api/openapi.html#tag/Virtual-Node-API/operation/put_virtualnode__virtualnodeshortid__cpi__targetcpifilechecksum_">`api/v1/virtualnode/<virtualnodeshortid>/cpi/<targetcpifilechecksum>` endpoint </a>.
 
 {{< note >}}
 The target CPI should have the same name, signer summary hash, and MGM group ID as the existing CPI. You can check the values of the existing CPI using 
-the <a href ="./retrieving.md">`/api/v5_1/virtualnode` endpoint </a>:
+the <a href ="./retrieving.md">`/api/v5_1/virtualnode` endpoint</a>.
```

Comment:
This link goues to [Retrieving Virtual Nodes](https://nadine.preview.docs.r3.com/en/platform/corda/5.0/deploying-operating/vnodes/retrieving.html) section (which btw has two titles so maybe you could remove one in this PR) and not the enpoind's API page: https://nadine.preview.docs.r3.com/en/platform/corda/5.0/reference/rest-api/openapi.html#tag/Virtual-Node-API/operation/get_virtualnode

Is that on purpose?